### PR TITLE
chore(app-structure): Add app structure

### PR DESCRIPTION
**Pull request by mwinel**
#### What does this PR do?
This PR adds a new file structure.

#### Description of Task to be completed
Restructure the app structure as below;
```
app/
  containers/
    home/
      home.jsx
    signup/
      signup.jsx
    login.jsx
  components/
    form/
      form.jsx
    buttons/
      round-button.jsx
      red-button.jsx
    signup.jsx
    login.jsx
  actions/
    signupAction.js
    loginAction.js
  reducers/
    signupReducer.js
    loginReducer.js
```

#### How should this be manually tested?
clone the repo
run `npm install`.
run `npm test`.
run `npm start`.

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#166510914](https://www.pivotaltracker.com/n/projects/2326199/stories/166510914)

#### Screenshots (if appropriate)
N/A

### CODE REVIEW

File: src/app/containers/Home.js [hunk 1]
```diff
@@ -1,11 +1,14 @@
 import React, { Component } from 'react';
+import Aux from '../hoc/Aux';
 
 class Home extends Component {
   render () {
     return (
-      <div>
-        <h5>Authors Haven</h5>
-      </div>
+      <Aux>
```
Comment (njeri-ngigi) [inline src/app/containers/Home.js]: I'm curious as to why you wrapped this in a hoc component

**Review by engjames — APPROVED** [review]
This file structure looks good to me as we discussed it with most the teammates.

File: src/app/containers/Home.js (hunk 1 above)
Comment (njeri-ngigi) [reply src/app/containers/Home.js]: If you have one element you can just return that.
In this case, you can return the h5 element. A hoc can be used when you have elements that are on the same level (i.e. not nested).

e.g.
```
<Aux>
    <Navbar/>
    <Home/>
    <Footer/>
</Aux>
```

**Review by njeri-ngigi — COMMENTED** [review]
- place the reducers and actions folder under the redux folder
- the Login folder looks misplaced. If its tests, you can have a tests folder or rather add a component and tests in one folder. e.g.

```
/containers
    /Home
        home.jsx
        home.test.js
```
or have a tests folder
e.g.
```
/app
    /__tests__
    /containers
    /components
   ```

**Review by joelmugaya — APPROVED** [review]

File: src/app/containers/Home.js (hunk 1 above)
Comment (KabohaJeanMark) [reply src/app/containers/Home.js]: Thanks for giving clarity about this @njeri-ngigi. I had been looking it up. Good work on the file structure Nelson. I am sure after you incorporate the few changes requested by our TTL we shall be able to rebase off develop and adopt this.
Like @engjames said, the file structure is as we discussed.

Comment (mwinel) [reply src/app/containers/Home.js]: I was looking at that we might have elements that are on the same level in the nearby future.

Comment (njeri-ngigi) [reply src/app/containers/Home.js]: I understand that. What I'm pointing out is how you used it here, it's not necessary here, you can just return the h5 element as it is

**Pull request merged**